lib/manageiq/api/common/request.rb

```diff
@@ -47,7 +47,21 @@ def user
         end
 
         def to_h
-          {:headers => headers.to_h, :original_url => original_url}
+          {:headers => forwardable, :original_url => original_url}
+        end
+
+        def forwardable
+          FORWARDABLE_HEADER_KEYS.each_with_object({}) do |key, hash|
```

Comment:
Any reason not to use the `.slice` from above?

Reply:
Yes.  It turns out that .to_h of an ActionDispatch::Http::Headers doesn't do what you'd expect.  We need to use the accessors instead.

Reply:
More accurately `ActionDispatch::Http::Headers.from_hash` doesn't do what you'd expect.  That is, from_hash expects a Hash of what look like ENV vars.  Methods are expecting that structure in order to work in a case-insensitive manner.  This is the best examples I could come up with:

```ruby
irb(main):003:0> h = ActionDispatch::Http::Headers.from_hash("x-foo" => "x")
=> #<ActionDispatch::Http::Headers:0x00007f92405a0760 @req=#<ActionDispatch::Request:0x00007f92405a0788 @env={"x-foo"=>"x"} ...>
irb(main):004:0> h["x-foo"]
=> nil
irb(main):005:0> h["HTTP_X_FOO"]
=> nil
irb(main):006:0> h = ActionDispatch::Http::Headers.from_hash("HTTP_X_FOO" => "x")
=> #<ActionDispatch::Http::Headers:0x00007f92405ae798 @req=#<ActionDispatch::Request:0x00007f92405ae7c0 @env={"HTTP_X_FOO"=>"x"} ...>
irb(main):007:0> h["x-foo"]
=> "x"
irb(main):008:0> h["HTTP_X_FOO"]
=> "x"
```